src/main/java/com/amazon/ion/impl/_Private_IonTextWriterBuilder.java

```diff
@@ -1,48 +1,20 @@
-/*
- * Copyright 2007-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
- *
- * Licensed under the Apache License, Version 2.0 (the "License").
- * You may not use this file except in compliance with the License.
- * A copy of the License is located at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * or in the "license" file accompanying this file. This file is distributed
- * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
- * express or implied. See the License for the specific language governing
- * permissions and limitations under the License.
- */
-
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
 package com.amazon.ion.impl;
 
-import static com.amazon.ion.impl._Private_Utils.initialSymtab;
 
-import com.amazon.ion.IonCatalog;
-import com.amazon.ion.IonSystem;
-import com.amazon.ion.IonWriter;
-import com.amazon.ion.SymbolTable;
-import com.amazon.ion.system.IonSystemBuilder;
 import com.amazon.ion.system.IonTextWriterBuilder;
 import com.amazon.ion.system.SimpleCatalog;
-import com.amazon.ion.util._Private_FastAppendable;
-import java.io.OutputStream;
 
 /**
+ * Contains configuration common to text writers for all Ion versions.
  * NOT FOR APPLICATION USE!
  */
-public class _Private_IonTextWriterBuilder
+public abstract class _Private_IonTextWriterBuilder
```

Comment:
This is a good suggestion, and I've applied it in two steps, corresponding two the two commits in this PR that can be viewed separately.
1. Adding a parameter to `_Private_IonTextWriterBuilder` for the concrete type, which enables some minor cleanups.
2. Adding a parameter to the existing public `IonTextWriterBuilder` abstract class, allowing the concrete type to be plumbed all the way through. This allows us to remove all of the cast-only overrides in `_Private_IonTextWriterBuilder_1_1`. The downside is that it will add warnings to existing code that assigns to variables with the type `IonTextWriterBuilder`, because they'll be missing the type parameter. I'm not sure whether we'd like to consider this a breaking change or not. The code still runs, as confirmed by the tests. We can always drop the second of the two commits if we don't want to move forward with this.

Reply:
I think adding a parameter to the existing public class/interface needs to be considered a breaking change. However, I think you can probably still get rid of the cast-only overrides if you make all of the methods in `IonTextWriterBuilder` abstract and move their implementations into `_Private_TextWriterBuilder<T>`.

I know that in general, that's even more breaking than the change you have made, but the javadoc for `IonTextWriterBuilder` does say:
> This class should not be extended by code outside of this library."

Reply:
I think I'll just drop the second of the two commits currently proposed. I agree that we should avoid adding a parameter so that we don't break builds that fail if warnings are encountered. Unfortunately getting rid of cast-only overrides isn't possible even if we make the IonTextWriterBuilder methods abstract because of the duplication (with different declared return types) between IonTextWriterBuilder/_Private_IonTextWriterBuilder (and its ancestors) and the new IonTextWriterBuilder_1_1 interface.